src/store-tree/index.ls

```diff
@@ -40,45 +37,34 @@ class StoreTree extends StoreNode
 
 
   $is-loading: ->
-    for own key of @_children when @[key].$is-loading!
-      return yes
-    no
+    @_children |> values |> any (.$is-loading!)
 
 
   $reset: !-> @$batch-emit-updates ~>
-    @_for-each-subnode (subnode) -> subnode.$reset!
+    @_children |> Obj.each (.$reset!)
 
 
   $set: (data) !-> @$batch-emit-updates ~>
     switch typeof! data
-    | \Object   => @_for-each-subnode (subnode, key) -> subnode.$set data[key] if data[key] isnt undefined
-    | \Null     => @_for-each-subnode (subnode) -> subnode.$set null
+    | \Object   => @_children |> obj-to-pairs |> each ([key, subnode]) -> subnode.$set data[key] if data[key] isnt undefined
```

Comment:
This code is complex and it creates a good amount of temp data, for a pretty common operation. I don't want to prematurely optimize for performance, but maybe using a loop comprehension is both simpler and faster here. Something like:

```ls
subnode.$set(data[key]) for key, subnode of @_children when data[key]
```